client/state/selectors/is-eligible-for-domain-to-paid-plan-upsell.js

```diff
@@ -18,8 +19,9 @@ const isEligibleForDomainToPaidPlanUpsell = ( state, siteId ) => {
 	const userCanManageOptions = canCurrentUser( state, siteId, 'manage_options' );
 	const siteIsOnFreePlan = isSiteOnFreePlan( state, siteId );
 	const siteHasMappedDomain = isMappedDomainSite( state, siteId );
+	const siteIsVipSite = isVipSite( state, siteId );
 
-	return userCanManageOptions && siteHasMappedDomain && siteIsOnFreePlan;
+	return userCanManageOptions && siteHasMappedDomain && ! siteIsVipSite && siteIsOnFreePlan;
```

Comment:
Purely a taste thing, but I think I'd break this out into a separate test - `if ( siteIsVipSite ) { return false; }`. It's the same logically, but it seems a little clearer in intent.

We could dedupe `userCanManageOptions` and `isVipSite` into a shared `isEligibleForUpsells` selector, so the two selectors only have the information they actually care about, but it's probably better to hold off on that. My own DRY threshold is generally 3 copies.